### ISSUE TITLE
Accessibility: single product missing main tag

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43572]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Relevant line of code: https://github.com/woocommerce/woocommerce/blob/ad735b2bf202f8a33971e4a1c242147ec274256c/plugins/woocommerce/templates/templates/blockified/single-product.html#L4

The template for the product single is missing a `<main>` tag which results in no skip links being added to the page.

In this template, the group that contains all the blocks has been set to use a <div>.
<img width="1167" alt="Screenshot showing group block set to div" src="https://github.com/woocommerce/woocommerce/assets/13153456/3da23343-959b-45d2-9150-7fdf1f805953">

### Expected behavior

There should be a <main> tag on every page as the wrapper for the content.

### Actual behavior

There's no main tag.

### Steps to reproduce

1. Set up a shop page in Twenty Twenty-Four.
2. Inspect the code on the front end.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by roykho** [comment]
@amberhinds - Thanks for reporting this. We'll have a look when available.
